client/checkout/classic/payment-processing.js

```diff
@@ -192,6 +192,16 @@ function createStripePaymentMethod(
 		};
 	}
 
+	if (
+		getUPEConfig( 'isOrderPay' ) &&
+		[ 'afterpay_clearpay', 'affirm' ].includes( paymentMethodType ) &&
+		params?.billing_details?.address &&
+		! params.billing_details.address.line1
+	) {
+		// These payment methods don't accept an address object with partial information.
+		delete params.billing_details.address;
+	}
```

Comment:
I was able to successfully place orders with Affirm and Afterpay by adding this code.

Although I was using test mode, the order seems to have all the correct information. I assume any missing details would be captured by Affirm/Afterpay?